moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js

```diff
@@ -59,18 +76,25 @@ const ComponentForm = ({
     GET_COMPONENTS_FORM_OPTIONS
   );
   const componentOptions = useComponentOptions(optionsData);
-  const { component } = watch();
+  const phaseOptions = usePhaseOptions(optionsData);
+  const { component, phase } = watch();
+  const subphaseOptions = useSubphaseOptions(phase?.data.moped_subphases);
   const internalTable = component?.data?.feature_layer?.internal_table;
   const [areSignalOptionsLoaded, setAreSignalOptionsLoaded] = useState(false);
   const onOptionsLoaded = () => setAreSignalOptionsLoaded(true);
 
   const subcomponentOptions = useSubcomponentOptions(component);
+  const [useComponentPhase, setUseComponentPhase] = useState(
+    !!initialFormValues?.component.moped_phase
```

Comment:
Nice! I was looking around for how the switch initializes in the on position if there is already a phase. This works perfect!